addon/components/plugins/text-style/highlight.ts

```diff
@@ -0,0 +1,51 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { SayController, Selection } from '@lblod/ember-rdfa-editor';
+
+type Args = {
+  controller: SayController;
+  defaultColor: string;
+};
+
+export default class HighlightMenu extends Component<Args> {
+  @tracked selectedColor = this.args.defaultColor;
+
+  get controller() {
+    return this.args.controller;
+  }
+
+  get schema() {
+    return this.controller.schema;
+  }
+
+  get state() {
+    return this.controller.activeEditorState;
+  }
+
+  @action
+  setHighlight(color: string) {
+    const mark = this.schema.marks.highlight.create({ value: color });
```

Comment:
would like to see this wrapped in a command, so users of custom buttons can easily replicate the behavior

Reply:
All right, I moved the functionality into a command. I'll now look into the ability to remove highlights.